src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -468,6 +470,25 @@ public IndexCoordinates getIndexCoordinatesFor(Class<?> clazz) {
 		return getRequiredPersistentEntity(clazz).getIndexCoordinates();
 	}
 
+	@Override
+	public <T> UpdateResponse update(T entity) {
+		return update(this.buildUpdateQueryByEntity(entity), this.getIndexCoordinatesFor(entity.getClass()));
```

Comment:
```suggestion
		return update(buildUpdateQueryByEntity(entity), getIndexCoordinatesFor(entity.getClass()));
```
we don't qualify method calls with `this.`.